File: Kotlin/practice/codes/2447.kt
import java.util.Scanner
import java.io.BufferedWriter
import java.io.OutputStreamWriter

var arr = Array(1,{CharArray(1)})

fun main() = with(Scanner(System.`in`)){
    val bw = BufferedWriter(OutputStreamWriter(System.`out`))
    
    val N = nextInt()
    arr = Array(N,{CharArray(N)})
    star(0,0,N,false)
    for(i in 0..N-1){
        for(j in 0..N-1){
            bw.write(arr[i][j].toString())
        }
        bw.newLine()
    }
    bw.flush()
    bw.close()
}

fun star(x: Int, y: Int, N: Int, blank: Boolean){
    if(blank){
    	for(i in x..x+N-1){
            for(j in y..y+N-1){
                arr[i][j] = ' '
            }
        }
        return
    }
    
    if(N == 1){
        arr[x][y] = '*'
        return
    }
    else{
        var size = N/3
        var count = 0
        
        for(i in x..x+N-1 step size){
            for(j in y..y+N-1 step size){
                count++
                if(count == 5){
                    star(i,j,size,true)
                }
                else{
                    star(i,j,size,false)
                }
            }
        }
    }
}

File: Algorithm/codes/2447.cpp
#include <iostream>
#include <vector>

using namespace std;

void star(vector<vector<char>> &arr, int x, int y, int size) {
	int cnt = 0;
	if (size == 1)
		arr[x][y] = '*';
	else {
		for (int i = x; i < x + size; i += size/3) {
			for (int j = y; j < y + size; j += size / 3) {
				cnt++;
				if (cnt != 5) star(arr, i, j, size / 3);
			}
		}
	}
}

int main() {
	int n;
	cin >> n;
	vector<vector<char>> arr(n, vector<char>(n, ' '));
	star(arr, 0, 0, n);
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++)
			cout << arr[i][j];
		cout << '\n';
	}
}
